use main frame url for webrequest first party url

diracdeltas opened the pull request:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan: covered by automated test
fix #4137
fix https://github.com/brave/browser-laptop/issues/4300

auditors: @bridiver

bridiver commented:
this looks good, but can we add some tests for it? The changes look right, but it's hard to know for sure without either automated or manual testing

diracdeltas commented:
@bridiver thanks, i have rebased and added a test which fails on master but works with this PR. the problem appears to be that `firstPartyUrl` in an cross-origin iframe is `data:` instead of the main frame URL, which is consistent with https://tools.ietf.org/html/draft-west-first-party-cookies-07#section-2.1.
